### PR TITLE
Detect multiple versions of the Hazelcast classes HZ-732

### PR DESCRIPTION
**Pull request by ldziedziul**
Added detection logic printing a warning if multiple `HazelcastInstance` class files are found. 

It's related to the issue with using `hazelcast-enterprise` together with `hazelcast-spring` which can result in having `hazelcast-enterprise` and `hazelcast` jars on the classpath (`hazelcast-enterprise` already contains shaded `hazelcast` ).

Related to https://hazelcast.atlassian.net/browse/HZ-732

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

**Review by frant-hartm — APPROVED** [review]

**Review by kwart — COMMENTED** [review]
Have you considered detection through standard classloader methods?
E.g.

```java
boolean hasMultipleNodeExtensions() throws IOException {
    Enumeration<URL> e = getClass().getClassLoader().getResources("META-INF/services/" + NodeExtension.class.getName());
    return Collections.list(e).size() > 1;
}
```

**Comment by ldziedziul** [comment]
> Have you considered detection through standard classloader methods?

@kwart well I think I've used the same classloader method but I've switched checking to a resource in `META-INF/services` :) https://github.com/hazelcast/hazelcast/pull/20148/commits/b2e6c3fd14849f1762d6a6721e271eded94e3ae0

File: hazelcast/src/test/java/com/hazelcast/instance/impl/DuplicatedResourcesScannerTest.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import java.nio.file.StandardCopyOption;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+
+public class DuplicatedResourcesScannerTest {
```
Comment (kwart) [inline hazelcast/src/test/java/com/hazelcast/instance/impl/DuplicatedResourcesScannerTest.java:41]: There are missing our lovely test class annotations. 
Sth like
```java
@RunWith(HazelcastParallelClassRunner.class)
@Category({QuickTest.class, ParallelJVMTest.class})
```

Comment (ldziedziul) [reply hazelcast/src/test/java/com/hazelcast/instance/impl/DuplicatedResourcesScannerTest.java:51]: Thanks, I've also noticed that there's was a bug in one of the tests (I forgot to use a classloader with a custom jar), both issues fixed here: https://github.com/hazelcast/hazelcast/pull/20148/commits/31e258e03c70e35d70fa220ab004857fd1df0c36

**Review by kwart — APPROVED** [review]

**Pull request merged**